components/membership/certificates-client-impl/src/main/kotlin/net/corda/membership/certificate/client/impl/HostedIdentityEntryFactory.kt

```
@@ -112,7 +112,7 @@ internal class HostedIdentityEntryFactory(
             logger.warn("Could not retrieve group policy for validating TLS trust root certificates.", e)
             null
         } ?: throw CordaRuntimeException("No group policy file found for holding identity ID [${nodeInfo.holdingIdentity.shortHash}].")
-        val sessionPublicKey = getKey(holdingIdentityShortHash.value, sessionKeyId)
+        val sessionPublicKey = getKey(holdingIdentityShortHash.value, sessionKeyId?.let { ShortHash.of(sessionKeyId) })
```

We still need to throw a 40x error if the sessionKeyId is not a valid short hash. The session key should be a short hash all the way up... (In the `CertificatesClientImpl` and then in the `NetworkRestResourceImpl` where we should parse it).

---

Done